addons/l10n_in/models/account_invoice.py

```diff
@@ -78,6 +190,118 @@ def _onchange_name_warning(self):
             }}
         return super()._onchange_name_warning()
 
+    def _l10n_in_calculate_aggregate_totals(self, partners):
+        def get_company_ids(company):
+            company_ids = company
+            if company.parent_id:
+                company_ids |= company.parent_id
+                if company.parent_id.child_ids:
+                    company_ids |= company.parent_id.mapped('child_ids')
+            if company.child_ids:
+                company_ids |= company.mapped('child_ids')
+            return company_ids
+
+        def get_move_lines_by_partner(partner_domain, start_date, end_date, account_ids, period):
+            company_ids = get_company_ids(self.company_id)
+            partner_ids = self.env['res.partner'].search(partner_domain)
+            query = """
+                SELECT line.id
+                FROM account_move_line line
+                JOIN account_move move ON line.move_id = move.id
+                JOIN account_account account ON line.account_id = account.id
+                WHERE move.date >= %s
+                    AND move.date <= %s
+                    AND move.company_id IN %s
+                    AND account.id IN %s
+                    AND account.l10n_in_tds_tcs_section IS NOT NULL
+                    AND account.l10n_in_tds_tcs_section IN (
+                        SELECT id
+                        FROM account_tax_group
+                        WHERE l10n_in_aggregate_period = %s
+                    )
+            """
+            params = [start_date, end_date, tuple(company_ids.ids), tuple(account_ids.ids), period]
+            if partner_ids:
+                query += " AND line.partner_id IN %s"
+                params.append(tuple(partner_ids.ids))
```

Comment:
```suggestion
                AND line.partner_id IN %s
            """
            params = [start_date, end_date, tuple(company_ids.ids), tuple(account_ids.ids), period, tuple(partnes.ids)]
```
@bhra-odoo are we considering any case without partner?

Reply:
@hamo-odoo If the partner is present, we need to consider the lines to check limits.